interview-questions.md

```diff
@@ -8,62 +8,69 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer:A parameter is the variable that is listed in the parentheses of the function, and when something is passed through the console.log (which is the argument) it is sent through that function.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The required parameters are the ones that need to be pushed from the argument. When an argument is passed, the map() function will alter the original array. 
 
-  Researched answer:
+  Researched answer: the array method called map(), calls a specified function for every array element and then creates a new array with that. This map() method does not change anything to the original array but creates a new array. We use this if we want to execute a different function to an array or an element within that array. 
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Map accesses all the elements within an array and filter, filters out specific elements in an array. 
 
-  Researched answer:
+  Researched answer: The filter() array method will filter out any unwanted elements within that array. The map() array method will transform every element in an array individually. 
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is a code block that makes an action and can be used multiple times on different data. A method is a built in function that can be used within a function. 
 
-  Researched answer:
+  Researched answer: A method is a function within an object or function.  If there is no method within the object, it is just an object. If there is no object surrounding the method, then it is a function. When a function is placed within the scope of an object, it is considered a method. 
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is when you can take a property from an object and apply it to a new variable. 
 
-  Researched answer:
+  Researched answer: Destructuring is when you take the properties from an object or the indexes of an array to separate them and to then create specific variables.
+
+  object(contains properties) --> destructure(property) --> create new variable(containing same property)
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting is related to the arrangement of code and how variables are accessed. 
 
-  Researched answer:
+  Researched answer: Hoisting is the process of moving the declaration of variables, functions or classes to the top of their scope prior to the execution of the code. It gives us the advantage that wherever the variables etc.. are declared, they are then moved to the top of their scope and does not matter if their scope is global or local. 
```

Comment:
Great job on this section.  Both your answers and researched answers were very well worded.